### PR TITLE
Handle edge cases in job detail [FLYW-649]

### PR DESCRIPTION
**Pull request by ehlertjd**
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and log messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

File: api/jobs/handlers.py [hunk 1]
```diff
@@ -624,6 +624,8 @@ def get_detail(self, _id):
         def get_container(ref):
             """Helper for cached retrieval of container"""
             # Normalize id
+            if not ref or not ref.id:
+                return None
```
Comment (hkethi002) [inline api/jobs/handlers.py]: Out of curiosity, how does this happen?

**Comment by nagem** [comment]
¯\\_(ツ)_/👍

File: api/jobs/jobs.py [hunk 2]
```diff
@@ -159,7 +159,7 @@ def load(cls, e):
         # Don't modify the map
         d = copy.deepcopy(e)
 
-        if d.get('inputs'):
+        if d.get('inputs') is not None:
```
Comment (nagem) [inline api/jobs/jobs.py]: I'm not sure I understand this one - you want to still run through this code path if inputs is `{}`?

Comment (nagem) [reply api/jobs/jobs.py]: Or were inputs empty lists sometimes?

Comment (hkethi002) [reply api/jobs/jobs.py]: It's `[]` in mongo and we still want to convert it into a dictionary

File: api/jobs/handlers.py (hunk 1 above)
Comment (ehlertjd) [reply api/jobs/handlers.py]: Thank you for asking! This was happening when the job destination was not set or None. However when I added the missing integration test I uncovered another NPE if inputs exist, but destination is `None` which is not something we observed in the database, but can't rule it out.

**Comment by ehlertjd** [comment]
@roycollins Good thought, but that already exists in the current codebase. This fix is just to handle ancient jobs which didn't have those restrictions.
